source/code/providers/Container_ContainerLog_Class_Provider.cpp

```diff
@@ -0,0 +1,288 @@
+/* @migen@ */
+#include <MI.h>
+#include "Container_ContainerLog_Class_Provider.h"
+
+#include <map>
+#include <stdio.h>
+#include <string>
+#include <string.h>
+#include <syslog.h>
+#include <time.h>
+#include <unistd.h>
+#include <uuid/uuid.h>
+#include <vector>
+
+#include "../cjson/cJSON.h"
+#include "../dockerapi/DockerRemoteApi.h"
+#include "../dockerapi/DockerRestHelper.h"
+
+#define LASTLOGQUERYTIMEFILE "/var/opt/microsoft/docker-cimprov/state/LastLogQueryTime.txt"
```

Comment:
Blech, a new one of these (we had another state file for time). I wonder if you should write a general state class (like SCX has for the logfilereader) that stores state in a proper way and manages what to do if said state information doesn't exist?

I thought the other time file was a bit of a cludge, having two seems twice as bad. It can stay, I guess, but you should think about doing something more formal.